src/Confluent.Kafka/Producer.cs

```diff
@@ -381,13 +380,16 @@ public long OutQueueLength
 
         /// <summary>
         ///     Wait until all outstanding produce requests, et.al, are completed. This should typically be done prior
-        //      to destroying a producer instance to make sure all queued and in-flight produce requests are completed
-        //      before terminating.
+        ///     to destroying a producer instance to make sure all queued and in-flight produce requests are completed
+        ///     before terminating. The wait is limited by the optional millisecondsTimeout parameter.
         /// </summary>
-        public void Flush(TimeSpan timeout)
```

Comment:
It is typical in the BCL to provide both TimeSpan and millisecondsTimeout overloads in analogous situations, so I'm in favor of leaving both. Eg
https://msdn.microsoft.com/en-us/library/system.threading.waithandle_methods(v=vs.110).aspx
https://msdn.microsoft.com/en-us/library/system.threading.thread_methods(v=vs.110).aspx